### PR TITLE
Adding Testing Module

### PR DESCRIPTION
**Pull request by Cmurilochem**
This PR adds the RSS Testing Module as per #8.

### CODE REVIEW

**Comment by Cmurilochem** [comment]
It may still require some content formatting after slides deck become available..

File: modules/testing/exercise1.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+c) To design the software's user interface and graphics.
+
+d) To accelerate the overall software development process.
+
+e) To facilitate easier collaboration among developers.
+
+---
+
+<small> correct answers: c) and b) </small>
```
Comment (JaroCamphuijsen) [inline modules/testing/exercise1.md:33]:
```suggestion
<small> **answers:** 
1.1 c) 
1.2 The most obvious answer is b), but in fact all answers except c) are correct
</small>
```

File: modules/testing/exercise1.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# Exercise 1
+
+Choose an alternative that correctly answers the following questions:
+
+**What is software testing?**
```
Comment (JaroCamphuijsen) [inline modules/testing/exercise1.md:5]:
```suggestion
**1.1 What is software testing?**
```

File: modules/testing/exercise1.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+c) The process of verifying and validating that a software meets the required specifications and works as intended.
+
+d) A step in the software lifecycle focused on community engagement and feedback.
+
+e) The practice of documenting software functionalities.
+
+---
+
+**Why is software testing important?**
```
Comment (JaroCamphuijsen) [inline modules/testing/exercise1.md:19]:
```suggestion
**1.2 Why is software testing important?**
```

File: modules/testing/exercise2.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
+# Exercise 2
```
Comment (JaroCamphuijsen) [inline modules/testing/exercise2.md:1]: Overall very nice. I think once we have the layout properly in place we can maybe format this exercise a bit better, for example as an actual table. Also, then, I don't think we need the empty "answers" part.

**Review by JaroCamphuijsen — APPROVED** [review]
Really nice first version of this module! In the future it would be good to add some content pages that cover the same content as in the slides but in a described/readable way and potentially some resources that should or could be read, with a time indication. See for example: https://esciencecenter-digital-skills.github.io/software-support-essentials/researchsoftware
Every time there is a "content" tab after a slides tab, which discusses more or less the same content and provides some reading material/resources. 
We can for now merge this pull request after you added the minor changes and I will put this in an issue for the future.

**Comment by JaroCamphuijsen** [comment]
For me this is fine to merge. #32 is the continuation of this PR

File: modules/testing/exercise2.md (hunk 4 above)
Comment (Cmurilochem) [reply modules/testing/exercise2.md:1]: Thanks! I agree!

**Pull request merged**